oak_attestation_verification/src/verifier.rs

```diff
@@ -465,8 +462,8 @@ fn verify_root_layer(
     endorsements: Option<&RootLayerEndorsements>,
     reference_values: &RootLayerReferenceValues,
 ) -> anyhow::Result<()> {
-    match values.report.as_ref() {
-        Some(Report::SevSnp(report_values)) => {
+    match (values.report.as_ref(), reference_values.amd_sev.as_ref(), reference_values.intel_tdx.as_ref(), reference_values.insecure.as_ref()) {
```

Comment:
I still think the logic here is not super clear. Would it be possible to match on the reference value only in the outer block, and then inside, for each reference value type, check that the evidence has the correct shape? I think that matches the intended semantics more closely.

Reply:
Fair enough, let's keep things as they are then, but personally I think we should change all of them for clarity (separately). WDYT?

Reply:
My take is that going by evidence was helpful because it's the only source describing an actual state while both reference value (in practice) and endorsements (theoretically) can cover multiple possibilities... But I'll let Tom and Conrad chime in on this, since they are familiar with the topic at a much deeper level